inc/saiport.h

```diff
@@ -1698,6 +1765,95 @@ typedef enum _sai_port_attr_t
      */
     SAI_PORT_ATTR_SYSTEM_PORT,
 
+    /**
+     * @brief MDIX status to the user
+     *
+     * @type sai_port_mdix_mode_status_t
+     * @flags READ_ONLY
+     */
+    SAI_PORT_ATTR_MDIX_MODE_STATUS,
+
+    /**
+     * @brief Configure MDIX mode
+     *
+     * @type sai_port_mdix_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_MDIX_MODE_AUTO
+     */
+    SAI_PORT_ATTR_MDIX_MODE,
+
+    /**
+     * @brief Configure Auto Neg mode
+     *
+     * @type sai_port_auto_neg_config_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_AUTO_NEG_CONFIG_MODE_DISABLED
+     */
+    SAI_PORT_ATTR_AUTO_NEG_CONFIG_MODE,
+
+    /**
+     * @brief Configure power mode
+     *
+     * @type sai_port_power_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_PORT_POWER_MODE_AUTO
+     */
+    SAI_PORT_ATTR_POWER_MODE,
+
+    /**
+     * @brief Enabled auto detection between 1000X and SGMII slave mode
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_PORT_ATTR_AUTO_DETECT,
```

Comment:
It looks like there is some dependency with the AUTO_MEDIUM attribute proposed below. For e.g, is this configurable when AUTO_MEDIUM is set to false?

Reply:
There is no relation/dependency between auto medium and auto detect. Auto-detect allows auto detection between SGMIi-Slave and 1000BaseX modules and Auto-medium allows both fiber and copper ports to be active at the same time. If auto-medium is set to false, then only copper medium is active and the value of auto-detect is ignored. Auto-detect value is only valid when SAI_PORT_ATTR_MEDIA_PREFERRED is set to true to allow Fiber medium to take precedence

Reply:
@SathishKumarTR-Dell  - please review/approve.  There is also a doc included in this PR.